examples/financial_document_analysis_with_llamaindex.ipynb

```diff
@@ -0,0 +1,554 @@
+    "Here, we pick *text-davinci-003* from OpenAI, allow unlimited output tokens.  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 26,
+   "id": "c4ec8b0a-d5fa-4f74-a2cc-5cc52e009bc6",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "llm = OpenAI(temperature=0, model_name=\"text-davinci-003\", max_tokens=-1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c2810e8c-1c88-49f5-aada-c49eccded166",
+   "metadata": {},
+   "source": [
+    "We construct a `ServiceContext` and set it as the global default, so all subsequent operations that depends on LLM calls will use the model we configured here."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "id": "05e016f9-2055-4885-8416-cc3aa2968242",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "service_context = ServiceContext.from_defaults(llm=llm)\n",
+    "set_global_service_context(service_context=service_context)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "71fddd07-ff4c-44d4-82af-64e2e416e853",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Data Loading and Indexing"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e3c1a778-fd5a-4249-9704-580c52cacb10",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "Now, we load and parse 2 PDFs (one for Uber 10-K in 2021 and another for Lyft 10-k in 2021).    \n",
+    "Under the hood, the PDFs are converted to plain text `Document` objects, separate by page.  \n",
+    "\n",
+    "> Note: this operation might take a while to run, since each document is more than 100 pages."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "id": "dd0ba028-1e70-4164-8af1-5f1df0ea76a9",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "lyft_docs = SimpleDirectoryReader(input_files=[\"data/10k/lyft_2021.pdf\"]).load_data()\n",
+    "uber_docs = SimpleDirectoryReader(input_files=[\"data/10k/uber_2021.pdf\"]).load_data()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 31,
+   "id": "d026ef11-ebc5-4ec3-9aab-8e065cd7f8a9",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Loaded lyft 10-K with 238 pages\n",
+      "Loaded Uber 10-K with 307 pages\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(f'Loaded lyft 10-K with {len(lyft_docs)} pages')\n",
+    "print(f'Loaded Uber 10-K with {len(uber_docs)} pages')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fd122d0d-2da6-4f46-aa2a-8a0049ad8694",
+   "metadata": {},
+   "source": [
+    "Now, we can build an (in-memory) `VectorStoreIndex` over the documents that we've loaded.  \n",
+    "\n",
+    "> Note: this operation might take a while to run, since it calls OpenAI API for computing vector embedding over document chunks."
```

Comment:
Does it do this in parallel or in sequence - if in sequence, can it done in parallel to reduce execution time? Also, what happens if some of the embeddings fail?

Reply:
It calls the endpoint in batches. And I think if some embeddings fail, the whole operate would fail. Definitely should add some better error catching here.